src/sync.js

```diff
@@ -352,77 +350,88 @@ class Sync {
   // If a file has been changed, we had to check what operation it is.
   // For a move, the first call will just keep a reference to the document,
   // and only at the second call, the move operation will be executed.
-  fileChanged (doc: Metadata, side: Side, rev: number, callback: Callback) {
+  async fileChangedAsync (doc: Metadata, side: Side, rev: number): Promise<void> {
     let from
     switch (true) {
       case doc._deleted && (rev === 0):
-        callback()
-        break
+        return
       case this.moveFrom != null:
         // $FlowFixMe
         from = (this.moveFrom: Metadata)
         this.moveFrom = null
         if (from.moveTo === doc._id) {
-          side.moveFile(doc, from, err => {
-            if (err) { this.moveFrom = from }
-            callback(err)
-          })
+          try {
+            await side.moveFileAsync(doc, from)
+          } catch (err) {
+            this.moveFrom = from
+            throw err
+          }
         } else {
           log.error('Invalid move')
           log.error(from)
           log.error(doc)
-          side.addFile(doc, function (err) {
-            if (err) { log.error(err) }
-            side.destroy(from, function (err) {
-              if (err) { log.error(err) }
-              callback(new Error('Invalid move'))
-            })
-          })
+          try {
+            await side.addFileAsync(doc)
+          } catch (err) {
+            log.error(err)
+            try {
+              await side.destroyAsync(from)
```

Comment:
The nesting seems wrong here. We should call `destroyAsync(from)` even if `addFileAsync(doc)` has suceeded